tsl/src/chunk_copy.c

```diff
@@ -418,6 +453,13 @@ chunk_copy_stage_create_empty_chunk(ChunkCopy *cc)
 
 	chunk_api_call_create_empty_chunk_table(ht, cc->chunk, NameStr(cc->fd.dest_node_name));
 
+	/*
+	 * Switch the empty chunk ownership to catalog owner for the lifetime of this operation.
+	 * This entire copy/move operation is run as superuser (including the remote connection
+	 * to the destination datanode), so the ALTER OWNER command will work without issues.
+	 */
+	chunk_copy_alter_chunk_owner(cc, NameStr(cc->fd.dest_node_name), false, false);
```

Comment:
I expected actually creating the chunk with superuser owner from the start, not requiring ALTERing before replication. But I guess this works because it is transactional, and an attacker won't see this chunk table until it is already altered to have the superuser owner?

Otherwise, there might still be a small window where the table exists with the owner that allows modifications.

Reply:
@erimatnor yes, that's correct. This is transactional so it's atomically visible as owned by superuser on creation.

Reply:
Same thoughts from me, I am also fine as long as this runs in the same transaction as the empty table creation for chunk and compressed chunks